packages/host/app/components/card-catalog/filters.gts

```diff
@@ -3,49 +3,54 @@ import Component from '@glimmer/component';
 import { BoxelDropdown, Menu } from '@cardstack/boxel-ui/components';
 import { MenuItem } from '@cardstack/boxel-ui/helpers';
 
-import { type RealmCards } from '@cardstack/runtime-common';
+import { RealmInfo } from '@cardstack/runtime-common';
 
 interface Signature {
-  availableRealms: RealmCards[];
-  selectedRealms: RealmCards[];
-  onSelectRealm: (realm: RealmCards) => void;
-  onDeselectRealm: (realm: RealmCards) => void;
+  availableRealms: Record<string, RealmInfo> | undefined;
+  selectedRealmUrls: string[];
+  onSelectRealm: (realmUrl: string) => void;
+  onDeselectRealm: (realmUrl: string) => void;
   disableRealmFilter: boolean;
 }
 
 export default class CardCatalogFilters extends Component<Signature> {
   get realmMenuItems() {
-    return this.args.availableRealms.map((realm) => {
-      return new MenuItem(realm.realmInfo.name, 'action', {
-        action: () => {
-          let isSelected = this.args.selectedRealms
-            .map((realm) => realm.url)
-            .includes(realm.url);
+    if (!this.args.availableRealms) {
+      return [];
+    }
+    return Object.entries(this.args.availableRealms).map(
+      ([realmUrl, realmInfo]) => {
+        return new MenuItem(realmInfo.name, 'action', {
+          action: () => {
+            let isSelected = this.args.selectedRealmUrls.includes(realmUrl);
```

Comment:
I changed the behavior of the realm filters because I think the previous behavior was confusing. By default, cards from all available realms were shown in the results while the selector didn't have any realms selected, which I think is a discrepancy:

Before (default state when opening the cards chooser, when all cards from all realms are shown):
<img width="327" alt="image" src="https://github.com/user-attachments/assets/9c45a0a0-779f-4993-b65b-ecfa3a1aa201">

After:
<img width="365" alt="image" src="https://github.com/user-attachments/assets/6f8af2df-af37-4688-8f21-07a8eaf77064">